src/lib/registration/ui/not_installed_products_dialog.rb

```diff
@@ -0,0 +1,196 @@
+    # but not registered products warning the user about it and allowing him
+    # to take some actions over them.
+    class NotInstalledProductsDialog
+      include Yast::Logger
+      include Yast::I18n
+      include Yast::UIShortcuts
+      include Yast
+
+      Yast.import "UI"
+      Yast.import "Popup"
+      Yast.import "Pkg"
+      Yast.import "Packages"
+      Yast.import "PackagesUI"
+
+      attr_accessor :registration, :registration_ui
+
+      def self.run
+        dialog = NotInstalledProductsDialog.new
+        dialog.run
+      end
+
+      def initialize
+        textdomain "registration"
+
+        self.registration = Registration.new(UrlHelpers.registration_url)
+        self.registration_ui = RegistrationUI.new(registration)
+      end
+
+      def run
+        Yast::Popup.Feedback(RegistrationUI::CONTACTING_MESSAGE, _("Checking product conflicts")) do
+          return :next if !registered_not_installed_addons?
+        end
+
+        Yast::UI.OpenDialog(Opt(:decorated), content)
+
+        begin
+          handle_dialog
+        ensure
+          Yast::UI.CloseDialog
+        end
+      end
+
+      private
+
+      def content
+        VBox(
+          MinWidth(80,
+            MinHeight(15,
+              VWeight(25, RichText(Id(:summary), not_installed_addons_summary))
+            )
+          ),
+          ButtonBox(
+            PushButton(Id(:cancel), Opt(:key_F9, :cancelButton), Yast::Label.AbortButton),
+            # FIXME: Maybe we could remove this option and just warn the user
+            PushButton(Id(:install), _("Ins&tall addons")),
+            PushButton(Id(:sync), _("&Deactive")),
+            PushButton(Id(:next), Opt(:okButton, :key_F10, :default), _("Continue"))
+          )
+        )
+      end
+
+      def handle_dialog
+        loop do
+          Yast::UI.SetFocus(:next)
+          ui = Yast::UI.UserInput
+          log.info "User input: #{ui}"
+          case ui
+          when :install
+            not_installed = install_products
+            # TRANSLATORS: Popup error showing all the addons that weren't
+            # installed, %s is the addons identifiers.
+            Yast::Popup.Error(_("These addons were not installed:\n\n%s") %
+                              not_installed.join("\n")) unless not_installed.empty?
+            update_summary
+          when :sync
+            registration_ui.synchronize_products(SwMgmt.installed_products)
+            update_summary
+          when :next, :cancel
+            return ui
+          end
+
+          return :next if !registered_not_installed_addons?
+        end
+      end
+
+      # Install the not installed products
+      # @return [Array] list of not installed (failed) products, empty list on success
+      def install_products
+        not_installed = []
+
+        Addon.registered_not_installed.each do |addon|
+          Yast::Popup.Feedback(RegistrationUI::CONTACTING_MESSAGE,
```

Comment:
I do not think we need yet another popup, normally there should be a package installation progress displayed. If not then we could just run the `inst_rpmcopy.rb` client instead of the `Yast::Pkg.PkgCommit(0)` call.